doc/qbk/07_concepts/BuffersGenerator.qbk

```diff
@@ -0,0 +1,128 @@
+[/
+    Copyright (c) 2016-2022 Vinnie Falco (vinnie dot falco at gmail dot com)
+
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+    Official repository: https://github.com/boostorg/beast
+]
+
+[section:BuffersGenerator BuffersGenerator]
+
+A [*BuffersGenerator] provides a generalized interface for generating
+serialized data for sequential processing.
+
+The generator will be asked to produce buffers. The consuming code will signal
+how much of the data has been consumed, and repeatedly query for buffers until
+no more data is available, or the generator indicates an error condition.
+
+In this way, serializers can be adapted as [*BuffersGenerator], for example
+__message_generator__ which provides a type-erased interface for a variety of
+concrete http message types.
+
+Overloads of `write` and `async_write` operations are provided as free
+functions. These operations will consume the output of a [*BuffersGenerator]
+and process the data by writing them to a __SyncWriteStream__ or
+__AsyncWriteStream__ respectively.
+
+[heading Associated Types]
+
+* [link beast.ref.boost__beast__core__is_buffers_generator `is_buffers_generator`]
+* __ConstBufferSequence__
+
+[heading Requirements]
+
+[warning
+    These requirements may undergo non-backward compatible
+    changes in subsequent versions.
+]
+
+In this table:
+
+* `G` denotes a type meeting the requirements of [*BuffersGenerator].
+* `g` denotes a value of type `G`.
```

Comment:
If an instance of _BuffersGenerator_ appears in code (such as in the refactored examples), I would use `g` as the variable name. Not `gen`, not `generator`, not `type_erased_message`, etc. I love single letter variables. I also love two character variables, but not quite as much as single letter variables. I don't love, but I do like three character variables. I think you see where I'm going with this.

Reply:
Yes I often use the longer formal parameter names in declarations because it looks better in documentation. There's nothing wrong with having full names in the declaration, and then using single letters in the definition.